File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const freq = {};
        const n = nums.length;

        for (let i=0; i<n; i++) {
            if (!freq[nums[i]]) {
                freq[nums[i]] = 1
            } else {
                freq[nums[i]]++
            }
        }

        const bucket = Array.from({ length: nums.length + 1 }, () => []);

        for (const key of Object.keys(freq)) {
            bucket[freq[key]].push(key)
        }
        console.log(bucket)
        const ans = [];
        let idx = n;
        while (ans.length < k) {
            ans.push(...bucket[idx--]);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zeros = 0
        idx = -1
        product = 1
        for i in range(len(nums)):
            if nums[i] == 0:
                zeros+=1
                idx=i
            else:
                product*= nums[i]
        if zeros > 1:
            return [0]*len(nums)
        if zeros == 1:
            ans = [0]*len(nums)
            ans[idx] = product
            return ans
        return [(product)//x for x in nums]